README.md

```diff
@@ -149,21 +156,27 @@ toybox = { "doll" => 'Sally' }
 
 Here is a good editor for regexp, maybe try and do it without using it first tho: https://rubular.com/
 
+I wasn't able to do this without the cheat sheet from that editor but this is my cheat assisted answer:
+
+`^\.(ts|tsx)(\.erb)?$`
+
+The problem was the first ? which allowed zero matches of (ts|tsx) (matching ., ..erb, .erb)
+
 #### Given the following file locations, what are the outputs?
 
 ```ruby
 # /User/gollum/precious/ring.rb
 File.dirname(__FILE__)
 ```
 
-- *give me answer*
+- 'precious'
```

Comment:
So for this one `__FILE__` is a ruby keyword and returns the name of the file being executed `ring.rb`. This can also include the directory if you are running it like:

```
ruby gollum/precious/ring.rb
```

in which case `__FILE__` will be `gollum/precious/ring.rb` and the answer `gollum/precious`.

Also if `__FILE__` is just the filename only you will get `..`

ruby-doc.org/docs/keywords/1.9/Object.html#method-i-__FILE__